packages/docusaurus-theme-classic/src/theme/hooks/useLogo.ts

```diff
@@ -9,12 +9,14 @@ import useThemeContext from '@theme/hooks/useThemeContext';
 import useBaseUrl from '@docusaurus/useBaseUrl';
 import isInternalUrl from '@docusaurus/isInternalUrl';
 import type {LogoLinkProps, useLogoReturns} from '@theme/hooks/useLogo';
-import useThemeConfig from '../../utils/useThemeConfig';
+import useThemeConfig, {NavbarLogo} from '../../utils/useThemeConfig';
 
 const useLogo = (): useLogoReturns => {
-  const {
-    navbar: {logo = {}},
-  } = useThemeConfig();
+  const {navbar} = useThemeConfig();
+
+  // TODO not clear if logo src is mandatory or not
```

Comment:
The `src` is only required by validation when parent `logo` object exist. If there is no `logo` in the navbar config there will be no warning. The current code mitigated this by setting the `logo` default value to an empty object which can be seen in the removed code above (`navbar: {logo = {}}`).